ui/app/ducks/gas/gas.duck.js

```diff
@@ -29,6 +31,7 @@ const initState = {
     fast: null,
   },
   basicEstimateIsLoading: true,
+  isEthGasPriceFetched: false,
```

Comment:
It seems that this introduces a competing state flag with 'basicEstimateIsLoading'. What I mean to say here is that in order to properly update state we have to check the difference between two keys instead of referencing one key.

What if we refactored the gas slice state to be more like:
```
basicEstimateStatus: 'LOADING' | 'READY' | 'FAILED';
```

In this way we can check one key and branch from the possible states instead of storing the 'status' of the state slice across a few different keys.

Reply:
As i'm reading more through the logic below I am wondering if this key doesn't actually mean what I think it does. It seems its only set to true if fetchExternalBasicGasEtimates fails and fetchEthGasPriceEstimates succeeds. Could you help me understand a little more?

Reply:
In mainnet, `isEthGasPriceFetched` flag is set to `true` only when the original [gas estimation service](https://api.metaswap.codefi.network/gasPrices) fails, but eth_gasprice is available (`fetchEthGasPriceEstimates` is called), and a warning message will be displayed based on this flag. In testnet the flag will be false, as testnets are not affected by the failure of this service.

When both the [gas estimation service](https://api.metaswap.codefi.network/gasPrices) and eth_gasprice fails, the basicEstimates being null is used to identify the condition, and the corresponding changes affects both mainnet and testnets.

Reply:
Alright so I might suggest the following to avoid using `null` as a logic check:

```
basicEstimateStatus: 'LOADING' | 'READY' | 'FAILED',
estimateSource: 'MetaSwaps' | 'eth_gasprice',
```

Now you can check for failed state by `basicEstimateStatus === 'FAILED'` and the source for your error message by `estimateSource === 'eth_gasprice'` and the code will be easier to maintain going forward. Thoughts on that?

Reply:
I have updated the PR based on these comments.